catalog/README.md

```diff
@@ -0,0 +1,37 @@
+# Catalog data for Ion test cases
+
+This directory includes Ion catalog data: shared symbol tables and modules that
+are reused across other Ion documents.
+
+Catalog entities are indexed by a name and version, where the name is a non-empty
+string (sequence of Unicode code points) and the version is a positive integer.
+
+Each Ion file in this directory contains one or more valid catalog entities.
+Each entity is represented as an Ion value following the canonical form of
+either:
+  * a shared symbol table: a struct with the first annotation 
+    `$ion_shared_symbol_table`
+  * a shared module: an S-expression with the first annotation 
+    `$ion_shared_module`
+
+Given a _name_ and _version_, implementations locate the relevant entity as 
+follows:
+
+  * First, look for a file path `<name>.ion` and seek the entity therein with
+    the given version.
+  * If no such file exists, seek the entity in the file `catalog.ion`. 
```

Comment:
Do you think we lose much if we get rid of one of these two options? As someone implementing this functionality, I'd appreciate having to implement only one of them.

Reply:
I worked through the details here to make sure there was a forward-compatible path should we want or need to use multiple files. I am anticipating future cases where there are large symbol tables that would be hard to manage all in one file.  The single `catalog.ion` file is likely to be fully loaded when accessed, potentially from every test file (depending on implementation), so I'm hoping that it stays relatively small.

That said, we only have to worry about this if we add such files.  I've not implemented this in the model framework, so you're fine ignoring the single-file clause for now.